TourMate/TourMate/Frontend/Plan/Views/PlanDiffView/PlanDiffView.swift

```diff
@@ -9,22 +9,44 @@ import SwiftUI
 
 @MainActor
 struct PlanDiffView<T: Plan>: View {
-    var viewModel: PlanViewModel<T>
+    @StateObject var leftViewModel: PlanViewModel<T>
+    @StateObject var rightViewModel: PlanViewModel<T>
+
+    @State private var leftVersion: Int
+    @State private var rightVersion: Int
+
+    let planDiffUtil: PlanDiffUtil
 
     init(planViewModel: PlanViewModel<T>) {
-        self.viewModel = planViewModel
+        let currentVersion = planViewModel.versionNumber
+
+        self._leftVersion = State(wrappedValue: planViewModel.isLatest ? currentVersion : currentVersion - 1)
+        self._rightVersion = State(wrappedValue: planViewModel.versionNumber)
```

Comment:
By default when we enter the PlanDiff the viewmodel is in the latest version, so both left and right views are currently using the latest version? I think the original logic was okay? Unless something there's something new

Reply:
oh previously i will set it to use the previous version. But this change somehow does not set the version

Reply:
Ok ya my bad haha the condition should be the other way around

Reply:
The comments view model seem to be re-creating itself though.. Trying to fix